modules/monitoring/cmk.tf

```diff
@@ -0,0 +1,63 @@
+# Creates/manages KMS CMK
+resource "aws_kms_key" "cmk" {
+  count = var.enable_cmk ? 1 : 0
+
+  description              = var.cmk_description
+  customer_master_key_spec = var.key_spec
+  is_enabled               = var.key_enabled
+  enable_key_rotation      = var.rotation_enabled
+  tags                     = var.tags
+  deletion_window_in_days  = var.deletion_window_in_days
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Id" : "kms-key-policy-access-control",
+    "Statement" : [
+      {
+        "Sid" : "Enable IAM User Permissions",
+        "Effect" : "Allow",
+        "Principal" : {
+          "AWS" : "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+        },
+        "Action" : "kms:*",
+        "Resource" : "*"
+      },
+      {
+        "Sid" : "Allow access for Key Administrators",
+        "Effect" : "Allow",
+        "Principal" : {
+          # Use 'var.sns_key_admin_arn' if available and root if not provided
+          "AWS" : var.sns_key_admin_arn != "" ? var.sns_key_admin_arn : "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
```

Comment:
@simeonzhekofff, is `root` okey here? I'm not familiar with the requirements.

Reply:
This parameter defines who is the principal of the key, It shouldn't be root. It's better to be a group for example KeyAdmins or something like that.